content/post/scaling-workloads/index.md

```diff
@@ -0,0 +1,116 @@
+to the CPU being significantly shorter. Just from the looks of it, this program may not be that complex
+like a compiler but there is always a chance this program will not run optimally. In fact, this program
+isn't as optimal as it can be for an x86 program. The most optimal is simply this.
+
+```asm
+mov ax, 1
+add ax, 1
+```
+
+We can further optimize this if we always assume `ax` will be `1` before we run the `add` instruction.
+This results in a program which looks like this.
+
+```asm
+add ax, 1
+```
+
+This program results in a single clock cycle and takes the least amount of execution time compared
+to the other variants of this program. However, the optimization of removing the `mov` to `ax` can
+be dangerous. If `ax` is not `1` then it would result in a different result being stored into `ax`.
+This is an example of an optimization which can be dangerous. It is considered dangerous as it
+is assuming the CPU state being a certain way with no checks. Adding checks can result in degrated
+performance. This is why many program which deals with high performance may use hand written assembly.
+The compiler may introduce certain optimizations such as the one in this example that either produces
+undefined behavior or is not optimal enough. Luckily, most people do not have to deal with having to
+understand what is going on. Modern compilers like GCC and LLVM's clang are optimal enough to prevent
+most issues. However, the thing which is important is the size of the workloads, having a good CPU for
+the workload, and enough RAM for the workload. Having the right CPU also matters as well, it is better
+to have a CPU for a specific architecture instead of cross compiling. This is because you have less
+to bootstrap for a specific CPU architecture. You typically already have what you need to compile for
+your native architecture.
+
+Having enough RAM depends on what the system typically does and how much you can parallelize really are
+the most important things here. This is going to become more apparant with the world moving away from x86
+and to ARM. ARM is quite different, it is known as a RISC (Reduced Instruction Set Computing) architecture.
+x86 is considered a CISC (Complex Instruction Set Computing) architecture. x86 has many more instructions
+compared to ARM. However, many instructions in a CISC architecture are similar to each other. This means
+you may have 30 different move instructions which each are optimized for a different data type. While on ARM,
+you have just what is needed. There can be just 5 move instructions on a RISC compared to the 30 in a CISC.
+This results in less complexity in the CPU which means less heat generated due to the less amount of silicon
+being used. This can result in a CPU operating colder and less prone to thermal throttling, it also requires
+less cooling. This can be optimal for laptops and phone, in fact this is why ARM is used in phones. In my
```

Comment:
`This can result in a CPU operating colder and less prone to thermal throttling` It *will* result in lower operating temperatures and thus reduced chance of thermal throttling.

`This can be optimal for laptops and phone, in fact this is why ARM is used in phones.` It *is* optimal for portable &/ battery powered devices, but again this section seems of questionable relevance to the topic of the post and seems to just be espousing the benefits of ARM.